op-e2e/actions/l2_validator_test.go

```diff
@@ -43,157 +39,72 @@ func TestValidatorBatchType(t *testing.T) {
+		submitAfterTransition = true
 	}
 
-	checkRightOutputSubmitted(t, sd.DeploymentsL1.L2OutputOracleProxy, miner, sequencer, seqEngine)
-}
+	// Assert validator submitted at least one output after transition
+	require.True(rt.t, submitAfterTransition)
 
-func proceedWithBlocks(t StatefulTesting, miner *L1Miner, sequencer *L2Sequencer, batcher *L2Batcher, batcherAddr common.Address, n int) {
-	for i := 0; i < n; i++ {
-		// L1 block
-		miner.ActEmptyBlock(t)
-		// L2 block
-		sequencer.ActL1HeadSignal(t)
-		sequencer.ActL2PipelineFull(t)
-		sequencer.ActBuildToL1Head(t)
-		// submit and include in L1
-		batcher.ActSubmitAll(t)
-		miner.includeL1Block(t, batcherAddr, 12)
-		// finalize the first and second L1 blocks, including the batch
-		miner.ActL1SafeNext(t)
-		miner.ActL1SafeNext(t)
-		miner.ActL1FinalizeNext(t)
-		miner.ActL1FinalizeNext(t)
-		// derive and see the L2 chain fully finalize
-		sequencer.ActL2PipelineFull(t)
-		sequencer.ActL1SafeSignal(t)
-		sequencer.ActL1FinalizedSignal(t)
-	}
+	checkRightOutputSubmitted(rt.t, rt.outputOracleContract, rt.sequencer, rt.seqEngine)
 }
 
-func submitOutput(t StatefulTesting, validator *L2Validator, miner *L1Miner) {
-	// submit to L1
-	validator.ActSubmitL2Output(t)
-	// include output on L1
-	miner.includeL1Block(t, validator.address, 12)
-	// Check submission was successful
-	receipt, err := miner.EthClient().TransactionReceipt(t.Ctx(), validator.LastSubmitL2OutputTx())
-	require.NoError(t, err)
-	require.Equal(t, types.ReceiptStatusSuccessful, receipt.Status, "submission failed")
-}
-
-func checkRightOutputSubmitted(t StatefulTesting, l2OOAddr common.Address, miner *L1Miner, sequencer *L2Sequencer, seqEngine *L2Engine) {
-	// check that L1 stored the expected output root
-	outputOracleContract, err := bindings.NewL2OutputOracle(l2OOAddr, miner.EthClient())
-	require.NoError(t, err)
-	// NOTE: If Proto Dank Sharding is introduced, the below code fix may be restored.
-	// block := sequencer.SyncStatus().FinalizedL2
-	// outputOnL1, err := outputOracleContract.GetL2OutputAfter(nil, new(big.Int).SetUint64(block.Number))
+// checkRightOutputSubmitted checks that L1 stored the expected output root
+func checkRightOutputSubmitted(t StatefulTesting, outputOracleContract *bindings.L2OutputOracle, sequencer *L2Sequencer, seqEngine *L2Engine) {
 	blockNum, err := outputOracleContract.LatestBlockNumber(&bind.CallOpts{})
 	require.NoError(t, err)
 	outputOnL1, err := outputOracleContract.GetL2OutputAfter(&bind.CallOpts{}, blockNum)
 	require.NoError(t, err)
 	block, err := seqEngine.EthClient().BlockByNumber(t.Ctx(), blockNum)
 	require.NoError(t, err)
 	require.Less(t, block.Time(), outputOnL1.Timestamp.Uint64(), "output is registered with L1 timestamp of L2 tx output submission, past L2 block")
-	outputComputed, err := sequencer.RollupClient().OutputAtBlock(t.Ctx(), blockNum.Uint64())
+	outputComputed, err := sequencer.RollupClient().OutputAtBlock(t.Ctx(), outputOnL1.L2BlockNumber.Uint64())
```

Comment:
Also a question, is there any reason that you changed from `blockNum` to `outputOnL1.L2BlockNumber`?

Reply:
I misunderstood that the `blockNum` is the latest block number of rollup client, not a value from `L2OutputOracle`. Since `blockNum` and `outputOnL1.L2BlockNumber` mean the same value, so I will revert it.